### ISSUE TITLE
Separate switch and outlet

### ISSUE THREAD

**Issue by hippopotam** [milo526/homebridge-tuya-web#35]
Currently, light switches are shown as outlets. Is it possible to separate them from outlets? I am using Treatlife devices with Smart Life. For some reason switch_accessory.js not working, so I changed In my copy of outlet_acessory.js OutletAccessory to SwitchAccessory to show my devices as switches.

Unfortunately, the plugs I have now are showing as switches, too.

Also, is it possible to change the icon of the switch to look like the actual wall switch, not the button?

**Comment by milo526** [comment]
They are probably exposed to homeassistant as outlets?

Could you follow the steps that check if the api supports your device as outlined in the reader and post the results here? Please indicate the name of the device and the expected device type.

**Comment by hippopotam** [comment]
Sure. Below is the results I got.
I added comments next to the name of the device to indicate device's real type, e.g. #<---- LIGHT SWITCH 
You will see that Light Switches and Outlets both are reported as switches. However, their respective icons are correct (switch and outlet)

Thanks!


% python3 debug_discovery.py
...
Getting devices
Got devices
{   'header': {'code': 'SUCCESS', 'payloadVersion': 1},
    'payload': {   'devices': [   {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay14906729310721fNjn/1568862727a52b550fe9a.jpg',
                                      'id': '34503056cc50e3ecbbae',
                                      'name': 'Kitchen Lights G'},  		# <--- LIGHT SWITCH (NEW)
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay14906729310721fNjn/1568862727a52b550fe9a.jpg',
                                      'id': '34503056bcddc2b085c3',
                                      'name': 'Kitchen Lights'},  		# <--- LIGHT SWITCH (NEW)
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay14906729310721fNjn/1568862445415ffab0175.jpg',
                                      'id': '25750750cc50e3726d96',
                                      'name': 'Dining Room Lights'},  	# <--- LIGHT SWITCH
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay14906729310721fNjn/1568862445415ffab0175.jpg',
                                      'id': '046617355002914b68cb',
                                      'name': 'Outside Lights'},  		# <--- LIGHT SWITCH
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay14906729310721fNjn/1568862445415ffab0175.jpg',
                                      'id': '25750750cc50e3727014',
                                      'name': 'Entry Light'},  				# <--- LIGHT SWITCH
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1519551146071pEnBd/1590981057603f7720ac8.png',
                                      'id': '10877013840d8e63a890',
                                      'name': 'Office'},  					# <--- OUTLET/PLUG
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1519551146071pEnBd/1590981057603f7720ac8.png',
                                      'id': '51721720c44f339ab068',
                                      'name': 'Salt Lamp'},  					# <--- OUTLET/PLUG
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1519551146071pEnBd/1590981057603f7720ac8.png',
                                      'id': '40840814c44f33999d40',
                                      'name': 'Candle Lamp'},  				# <--- OUTLET/PLUG
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1519551146071pEnBd/1590981057603f7720ac8.png',
                                      'id': '04713743c44f33adb7a8',
                                      'name': 'Sofa Light'},  				# <--- OUTLET/PLUG
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'scene',
                                                  'color_temp': 36294,
                                                  'online': False,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1515643003489ZiFAU/15731856298eb401cd167.jpg',
                                      'id': '40100577c44f33ba72ca',
                                      'name': 'Bed Light'}					 # <--- RGB BULB
                                  {   'data': {   'brightness': '670',
                                                  'color_mode': 'white',
                                                  'color_temp': 18224,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1515643003489ZiFAU/15731856298eb401cd167.jpg',
                                      'id': '40100577c44f33ba72cd',
                                      'name': 'Lanai Light'},					# <--- RGB BULB
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'scene',
                                                  'color_temp': 36294,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1515643003489ZiFAU/15731856298eb401cd167.jpg',
                                      'id': '40100577c44f33ba87f0',
                                      'name': 'Curio'},  								# <--- RGB BULB
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'white',
                                                  'color_temp': 36294,
                                                  'online': False,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaus.com/smart/icon/ay1515643003489ZiFAU/15731856298eb401cd167.jpg',
                                      'id': '40100577c44f33ba737a',
                                      'name': 'Desk Lamp'},						# <--- RGB BULB
                    ],
                   'scenes': []}}

**Comment by milo526** [comment]
So sadly this is not possible in a nice way since Tuya wrongfully exposes them.
There is however a work-around added to the latest beta version.

Please take a look at [Overruling Device Types in the README](https://github.com/milo526/homebridge-tuya-web/tree/v0.4.1#overruling-device-types).

You should add 

```json
{
  "options": {
    ....
  },
  "defaults": [
    {
      "id": "10877013840d8e63a890",
      "device_type": "outlet"
    }
  ]
```

To your config file.

To install the beta versions please install the `@milo526/homebridge-tuya-web@beta` package on your homebridge system.

**Comment by hippopotam** [comment]
So, default device type would be the switch, and we can override device type to outlet if device happens to be a plug/outlet.. I am pleased to confirm it works beautifully. Thank you!

Now, here's a repeating tidbit in the log that says this:

[TuyaWebPlatform] Could not find accessory in dictionary (9cc9b3df-dea0-48e7-98ec-8d99b9b406a5)

[TuyaWebPlatform] RatelimitError: Requesting too quickly.
    at TuyaWebApi.discoverDevices (/usr/local/lib/node_modules/@milo526/homebridge-tuya-web/src/TuyaWebApi.ts:159:32)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

**Comment by hippopotam** [comment]
Okay, after few hours, I've started experiencing problem: Home.app on my Mac mini is running fine, but on my iPhone, when I open the app, all devices, both tuya and non-tuya, show "updating" for several seconds, and then give "no response" error. If I uninstall beta and reinstall 0.33, there are no problems at all. It seems beta version interferes with homebridge somehow..

**Comment by milo526** [comment]
Could you please enable debug output on your homebridge instance and see if you get any useful logs from there?

**Comment by hippopotam** [comment]
Did that - same errors, nothing extra. However, if I remove polling interval parameter line completely, errors disappear.

**Comment by milo526** [comment]
Please be aware that continuing to send requests to the Tuya API (i.e. by opening the Home.app) will lengthen the time you get the timeout. Try ignoring the home app for 30 minutes and that should drastically improve the response rate.

**Comment by hippopotam** [comment]
Understood. Still, with older version 0.33, there’s no such problem. I see you packaged the Beta plugin different way than older plugin - could it be the reason, or part of?

**Comment by milo526** [comment]
The problem is very much still there on version 0.3.3 we just choose not to show the errors. The change with the beta is that the plugin is largely rewritten in TypeScript, the change from javascript to typescript has nothing to do with changes in requests being send.

**Comment by milo526** [comment]
This issue has been fixed in the [v0.4.2 release](https://github.com/milo526/homebridge-tuya-web/releases/tag/v0.4.2) if you are still experiencing this issue, or are experiencing other issues, please open a new bug report.

Thank you once again for this bug report, reports like these help make the plugin better!
